pkg/config/node.go

```diff
@@ -430,3 +437,45 @@ func GetClusterNameAndDomain(kubeconfigPath, clusterConfigPath string) (clusterN
 
 	return
 }
+
+func PopulateNodeAddresses(kubeconfigPath string, node *Node) {
+	// Get node list
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		log.Errorf("Failed to build client config: %s", err)
+		return
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Errorf("Failed to create client: %s", err)
+		return
+	}
+	nodes, err := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("Failed to get node list: %s", err)
+		return
+	}
+	for _, n := range nodes.Items {
+		name := ""
+		addr := net.IP{}
+		for _, a := range n.Status.Addresses {
+			if a.Type == v1.NodeHostName {
+				// We only want the shortname
+				name = strings.Split(a.Address, ".")[0]
+			}
```

Comment:
Wouldn't it be better to change it to 'if else' ?

Reply:
Sure, I can do that.

Reply:
Thanks